src/boot/shim.c

```diff
@@ -30,15 +30,44 @@ struct ShimLock {
         EFI_STATUS __sysv_abi__ (*read_header) (void *data, uint32_t datasize, void *context);
 };
 
+typedef struct {
+        EFI_STATUS (EFIAPI *LoadImage)(
+                bool BootPolicy,
+                EFI_HANDLE ParentImageHandle,
+                EFI_DEVICE_PATH *DevicePath,
+                void *SourceBuffer,
+                size_t SourceSize,
+                EFI_HANDLE *ImageHandle);
+        EFI_STATUS (EFIAPI *StartImage)(
+                EFI_HANDLE ImageHandle,
+                size_t *ExitDataSize,
+                char16_t **ExitData);
+        EFI_STATUS (EFIAPI *Exit)(
+                EFI_HANDLE ImageHandle,
+                EFI_STATUS ExitStatus,
+                size_t ExitDataSize,
+                char16_t *ExitData);
+        EFI_STATUS (EFIAPI *UnloadImage)(EFI_HANDLE ImageHandle);
+} ShimImageLoader;
+
 #define SHIM_LOCK_GUID \
         { 0x605dab50, 0xe046, 0x4300, { 0xab, 0xb6, 0x3d, 0xd8, 0x10, 0xdd, 0x8b, 0x23 } }
 
+#define SHIM_IMAGE_LOADER_GUID \
+        { 0x1f492041, 0xfadb, 0x4e59, { 0x9e, 0x57, 0x7c, 0xaf, 0xe7, 0x3a, 0x55, 0xab } }
+
 bool shim_loaded(void) {
         struct ShimLock *shim_lock;
 
         return BS->LocateProtocol(MAKE_GUID_PTR(SHIM_LOCK), NULL, (void **) &shim_lock) == EFI_SUCCESS;
 }
 
+bool shim_loader_available(void) {
+        ShimImageLoader *shim_image_loader;
```

Comment:
define as `void*`? then we do not have to carry the definition of ShimImageLoader